cloud/aws/alb/monitors-alb.tf

```diff
@@ -0,0 +1,203 @@
+		notifications         = coalescelist(var.heartbeat_notifications, var.notifications)
+		parameterized_subject = "[{{ruleSeverity}}]{{{detectorName}}} {{{ruleName}}} on {{{dimensions}}}"
+	}
+}
+
+resource "signalfx_detector" "no_healthy_instances" {
+	name = "${join("", formatlist("[%s]", var.prefixes))}[${var.environment}] ALB healthy instances"
+
+	program_text = <<-EOF
+		A = data('HealthyHostCount', filter=filter('namespace', 'AWS/ApplicationELB') and filter('stat', 'lower') and (not filter('AvailabilityZone', '*')) and ${module.filter-tags.filter_custom})${var.no_healthy_instances_aggregation_function}
+		B = data('UnHealthyHostCount', filter=filter('namespace', 'AWS/ApplicationELB') and filter('stat', 'upper') and (not filter('AvailabilityZone', '*')) and ${module.filter-tags.filter_custom})${var.no_healthy_instances_aggregation_function}
+		signal = (A/(A+B)).scale(100).${var.no_healthy_instances_transformation_function}(over='${var.no_healthy_instances_transformation_window}')
+		detect(when(signal < ${var.no_healthy_instances_threshold_critical})).publish('CRIT')
+		detect(when(signal < ${var.no_healthy_instances_threshold_warning})).publish('WARN')
+	EOF
+
+	rule {
+		description           = "are too low < ${var.no_healthy_instances_threshold_critical}"
+		severity              = "Critical"
+		detect_label          = "CRIT"
+		disabled              = coalesce(var.no_healthy_instances_disabled_critical, var.no_healthy_instances_disabled, var.detectors_disabled)
+		notifications         = coalescelist(var.no_healthy_instances_notifications_critical, var.no_healthy_instances_notifications, var.notifications)
+		parameterized_subject = "[{{ruleSeverity}}]{{{detectorName}}} {{{ruleName}}} ({{inputs.signal.value}}) on {{{dimensions}}}"
+	}
+
+	rule {
+		description           = "are too low < ${var.no_healthy_instances_threshold_warning}"
+		severity              = "Warning"
+		detect_label          = "WARN"
+		disabled              = coalesce(var.no_healthy_instances_disabled_warning, var.no_healthy_instances_disabled, var.detectors_disabled)
+		notifications         = coalescelist(var.no_healthy_instances_notifications_warning, var.no_healthy_instances_notifications, var.notifications)
+		parameterized_subject = "[{{ruleSeverity}}]{{{detectorName}}} {{{ruleName}}} ({{inputs.signal.value}}) on {{{dimensions}}}"
+	}
+
+}
+
+resource "signalfx_detector" "latency" {
+	name = "${join("", formatlist("[%s]", var.prefixes))}[${var.environment}] ALB latency"
+
+	program_text = <<-EOF
+		from signalfx.detectors.aperiodic import aperiodic
+		signal = data('TargetResponseTime', filter=filter('namespace', 'AWS/ApplicationELB') and filter('stat', 'mean') and filter('TargetGroup', '*') and (not filter('AvailabilityZone', '*')) and ${module.filter-tags.filter_custom})${var.latency_aggregation_function}.${var.latency_transformation_function}(over='${var.latency_transformation_window}')
```

Comment:
As I understand it uses an avergage on target groups, may be a max will be better to check highest latency